modules/structs/commit_status.go

```diff
@@ -5,39 +5,34 @@ package structs
 
 // CommitStatusState holds the state of a CommitStatus
 // It can be "pending", "success", "error" and "failure"
-type CommitStatusState string
+type CommitStatusState int
 
 const (
-	// CommitStatusPending is for when the CommitStatus is Pending
-	CommitStatusPending CommitStatusState = "pending"
-	// CommitStatusSuccess is for when the CommitStatus is Success
-	CommitStatusSuccess CommitStatusState = "success"
 	// CommitStatusError is for when the CommitStatus is Error
-	CommitStatusError CommitStatusState = "error"
+	CommitStatusError CommitStatusState = iota + 1
 	// CommitStatusFailure is for when the CommitStatus is Failure
-	CommitStatusFailure CommitStatusState = "failure"
+	CommitStatusFailure
+	// CommitStatusPending is for when the CommitStatus is Pending
+	CommitStatusPending
+	// CommitStatusSuccess is for when the CommitStatus is Success
+	CommitStatusSuccess
 )
 
-var commitStatusPriorities = map[CommitStatusState]int{
-	CommitStatusError:   0,
-	CommitStatusFailure: 1,
-	CommitStatusPending: 2,
-	CommitStatusSuccess: 3,
+var commitStatusNames = map[CommitStatusState]string{
+	CommitStatusError:   "error",
+	CommitStatusFailure: "failure",
+	CommitStatusPending: "pending",
+	CommitStatusSuccess: "success",
+}
+
+// String returns the string name of the CommitStatusState
+func (css CommitStatusState) String() string {
+	return commitStatusNames[css]
 }
 
 // NoBetterThan returns true if this State is no better than the given State
-// This function only handles the states defined in CommitStatusPriorities
 func (css CommitStatusState) NoBetterThan(css2 CommitStatusState) bool {
```

Comment:
This requires checking for special values, enhancing the robustness of the function, and avoiding issues caused by incorrect usage by the caller, as seen in #26121.

Reply:
Finally, I want to remove this function. 
It is only used in `MergeRequiredContextsCommitStatus`. But have no good idea about how to rewrite it now.
Or maybe I will add `CommitStatusState.IsValid` which should be called before we compare them.